src/sentry/models/event.py

```diff
@@ -94,36 +95,61 @@ def generate_node_id(cls, project_id, event_id):
         """
         return md5('{}:{}'.format(project_id, event_id)).hexdigest()
 
-    # TODO (alex) We need a better way to cache these properties.  functools32
-    # doesn't quite do the trick as there is a reference bug with unsaved
-    # models. But the current _group_cache thing is also clunky because these
```

Comment:
Do we no longer have an issue with unsaved models?

Reply:
Moving the `lru_cache` to a static function on the project id (no `self` argument) solves it. The problem came from the cache using `self` as a cache key, which depended on `hash(self)`, which, for django models changes when you save the model.

However this PR has other potential problems related to cache invalidation, so I'm going to do a slightly different approach